Scripts/Runtime/Entities/TaskSystem/AbstractTaskDriver.cs

```diff
@@ -53,34 +56,40 @@ public abstract class AbstractTaskDriver : AbstractAnvilBase
         /// <summary>
         /// Reference to the associated <see cref="AbstractTaskSystem"/>
         /// </summary>
-        public AbstractTaskSystem TaskSystem { get; }
+        //TODO: #100 - Might be able to change this back so we can access from the outside
+        internal AbstractTaskSystem TaskSystem { get; }
 
         /// <summary>
         /// Reference to the associated <see cref="World"/>
         /// </summary>
         public World World { get; }
 
-        internal CancelRequestsDataStream CancelRequestsDataStream { get; }
-        internal List<AbstractTaskStream> TaskStreams { get; }
-        internal TaskDriverCancellationPropagator CancellationPropagator { get; private set; }
+        internal List<AbstractTaskDriver> SubTaskDrivers { get; }
+        
+        internal TaskDriverCancelFlow CancelFlow { get; }
+        internal TaskData TaskData { get; }
+        internal AbstractTaskDriver Parent { get; }
 
-        protected AbstractTaskDriver(World world, Type systemType)
+        protected AbstractTaskDriver(World world, Type systemType, AbstractTaskDriver parent)
         {
             //We can't just pull this off the System because we might have triggered it's creation via
             //world.GetOrCreateSystem and it's OnCreate hasn't occured yet so it's World is still null.
             World = world;
+            Parent = parent;
 
             TaskSystem = (AbstractTaskSystem)world.GetOrCreateSystem(systemType);
             Context = TaskSystem.RegisterTaskDriver(this);
 
-            m_SubTaskDrivers = new List<AbstractTaskDriver>();
-            TaskStreams = new List<AbstractTaskStream>();
+            SubTaskDrivers = new List<AbstractTaskDriver>();
             m_JobConfigs = new List<AbstractJobConfig>();
 
-            TaskStreamFactory.CreateTaskStreams(this, TaskStreams);
-            CancelRequestsDataStream = new CancelRequestsDataStream();
-
-            TaskDriverFactory.CreateSubTaskDrivers(this, m_SubTaskDrivers);
+            TaskData = new TaskData(this, TaskSystem);
+            CancelFlow = new TaskDriverCancelFlow(this, Parent?.CancelFlow);
+            TaskDriverFactory.CreateSubTaskDrivers(this, SubTaskDrivers);
+            
+            //MIKE - Ordering Question, not sure if there is a better flow
```

Comment:
I'm not sure if there's a better way to do this. 

A `TaskDriver` can have a parent or if null is a top level TaskDriver. We're guaranteed this now by the constructor arg. 

But a `CancelFlow` needs a parent as well and it's not quite the same mapping because while a TaskSystem can have multiple TaskDrivers, a `TaskDriverCancelFlow` must have a unique `SystemCancelFlow`. There are N `SystemCancelFlows` for every `TaskDriver` a `TaskSystem` has.

Which means, I need to get reference to the parent CancelFlow when creating the sub task drivers and I need the whole chain to be complete in order to build up the request data. 

Writing it out, I'm not sure there is a solution, but it just feels gross.

Reply:
Ok thanks